File: Section5 _ViewModel_LiveDataWithDataBinding/CodingChallenge1_java/app/src/main/java/com/comaye/codingchallenge1_java/MainActivity.java
package com.comaye.codingchallenge1_java;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.Observable;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import android.os.Bundle;
import android.view.View;
import com.comaye.codingchallenge1_java.databinding.ActivityMainBinding;
//---------------------------------------------------------------------------------
public class MainActivity extends AppCompatActivity {
ActivityMainBinding binding;
MainActivityViewModel viewModel ;
//----------------------------------------
@Override
protected void onCreate(Bundle savedInstanceState){
    super.onCreate(savedInstanceState);

    viewModel   =  new ViewModelProvider(this).get(MainActivityViewModel.class);
    viewModel.getCurrentCount();
    binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
    binding.setViewmodel(viewModel);
    MutableLiveData<Integer> myLivedata = viewModel.countLiveData;

    myLivedata.observe(this, new Observer<Integer>() {
        @Override
        public void onChanged(Integer integer){
            binding.txtNumber.setText(String.valueOf(integer));
        }
    });

}
//---------------------------------------------------------------------
}

File: Section4_ LiveData/CodingChallenge1/app/src/main/java/com/comaye/codingchallenge1/MainActivity.kt
package com.comaye.codingchallenge1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import com.comaye.codingchallenge1.databinding.ActivityMainBinding
import androidx.databinding.DataBindingUtil.setContentView
import androidx.lifecycle.Observer
//------------------------------------------------------------------------------------------------------
class MainActivity : AppCompatActivity() {
//------------------------------------------------------------------------------------------------------
private lateinit var binding: ActivityMainBinding
    private lateinit var viewModel: MainActivityViewModel
    private lateinit var viewModelFactory: MainActivityViewModelFactory
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding =   setContentView(this, R.layout.activity_main)
        viewModelFactory = MainActivityViewModelFactory(0)
        viewModel = ViewModelProvider(this,viewModelFactory).get(MainActivityViewModel::class.java)

        viewModel.countData.observe(this, Observer {
            binding.countText.text = it.toString()
        })

        binding.button.setOnClickListener {
            viewModel.updateCount()
        }
    }
//------------------------------------------------------------------------------------------------------
}
//------------------------------------------------------------------------------------------------------